%\VignetteIndexEntry{MetaGxPancreas: a package for pancreas cancer gene expression analysis}
%\VignetteDepends{xtable}
%\VignetteSuggests{}
%\VignetteKeywords{}
%\VignettePackage{MetaGxPancreas}

\documentclass[11pt]{article}

\usepackage[utf8]{inputenc}
\usepackage{authblk}
\usepackage{color}


\title{MetaGxOvarian: a package for ovarian cancer gene expression analysis}

\author[1]{Michael Zon}
\author[1]{Vandana Sandhu}
\author[1,2]{Benjamin Haibe-Kains\thanks{benjamin.haibe.kains@utoronto.ca }}

\affil[1]{Bioinformatics and Computational Genomics Laboratory, Princess Margaret Cancer Center, University Health Network, Toronto, Ontario, Canada}
\affil[2]{Department of Medical Biophysics, University of Toronto, Toronto, Canada}

\SweaveOpts{highlight=TRUE, tidy=TRUE, keep.space=TRUE, keep.blank.space=FALSE, keep.comment=TRUE}

<<setup,echo=FALSE,results=hide,eval=FALSE>>=
options(keep.source=TRUE, width = 50)
@

\begin{document}
\setkeys{Gin}{width=1.2\textwidth}
\SweaveOpts{concordance=TRUE}

\maketitle
\tableofcontents

%------------------------------------------------------------
\section{Installing the Package}
%------------------------------------------------------------

The MetaGxPancreas package is a compendium of Pancreatic Cancer datasets.
The package is publicly available and can be installed from Bioconductor into R version 3.5.0 or higher.
Currently, the phenoData for the datasets is overall survival status and overall survival time.
This survival information is available for 11 of the 15 datasets.

To install the MetaGxPancreas package from Bioconductor:
<<install-pkg,eval=FALSE>>=
source("http://bioconductor.org/biocLite.R")
biocLite("MetaGxPancreas")
@

%------------------------------------------------------------
\section{Loading Datasets}
%------------------------------------------------------------
First we load the MetaGxPancreas package into the workspace.

To load the packages into R, please use the following commands:
<<loadlib>>=
library(MetaGxPancreas)
esets = MetaGxOvarian::loadPancreasEsets()[[1]]
@

This will load 16 expression datasets. Users can modify the parameters of the function to restrict datasets that do not meet certain criteria for loading. Some example parameters are shown below:

\begin{description}
\item Datasets:  Retain only genes that are common across all platforms loaded (default = FALSE)
\item Datasets:  Retain studies with a minimum sample size (default = 0)
\item Datasets:  Retain studies with a minimum number of genes (default = 0)
\item Datasets:  Retain studies with a minimum number of survival events (default = 0)
\item Datasets:  Remove duplicate samples (default = TRUE)
\end{description}


%------------------------------------------------------------
\section{Obtaining Sample Counts in Datasets}
%------------------------------------------------------------

To obtain the number of samples per dataset, run the following:

<<sampleNumber-summary, results=tex>>=
numSamples <- vapply(seq_along(esets), FUN=function(i, esets){
  ncol(esets[[ii]]@assayData$exprs)
  }, numeric(1), esets=esets)


SampleNumberSummaryAll <- data.frame(NumberOfSamples = numSamples,
                                     row.names = names(esets))
total <- sum(SampleNumberSummaryAll[,"NumberOfSamples"])
SampleNumberSummaryAll <- rbind(SampleNumberSummaryAll, total)
rownames(SampleNumberSummaryAll)[nrow(SampleNumberSummaryAll)] <- "Total"

require(xtable)
print(xtable(SampleNumberSummaryAll,digits = 2), floating = FALSE)
@

%------------------------------------------------------------
\section{Session Info}
%------------------------------------------------------------
<<sessionInfo,echo=FALSE,results=tex,eval=TRUE>>=
toLatex(sessionInfo())
@

\end{document}
